src/gil.rs

```diff
@@ -494,6 +494,31 @@ impl EnsureGIL {
     }
 }
 
+/// Value with concurrent access protected by the GIL.
+///
+/// This is a synchronization primitive based on Python's global interpreter lock (GIL).
+/// It ensures that only one thread at a time can access the inner value via shared references.
+/// It can be combined with interior mutability to obtain mutable references.
+pub struct GILProtected<T> {
+    value: T,
+}
+
+impl<T> GILProtected<T> {
+    /// Place the given value under the protection of the GIL.
+    pub const fn new(value: T) -> Self {
+        Self { value }
+    }
+
+    /// Gain access to the inner value by giving proof of having acquired the GIL.
+    pub fn with_gil<'py>(&'py self, _py: Python<'py>) -> &'py T {
+        &self.value
+    }
+}
+
+unsafe impl<T> Send for GILProtected<T> where T: Send {}
```

Comment:
I don't think this implementation is needed, the only field is `T` so I would assume that auto impl will cover this.

Reply:
Certainly, this was mindless cargo-culting from `Mutex`.